File: Programming Basics with Python - november 2023/Tasks 2/sum_of_two_numbers.py
start_of_interval = int(input())
end_of_interval = int(input())
magic_number = int(input())
combination_found = False
combination_counter = 0

for i in range(start_of_interval, end_of_interval + 1):
    for j in range(start_of_interval, end_of_interval + 1):
        combination_counter += 1
        if i + j == magic_number:
            print(f"Combination N:{combination_counter} ({i} + {j} = {magic_number})")
            combination_found = True
            break
    if combination_found:
        break
else:
    print(f"{combination_counter} combinations - neither equals {magic_number}")

File: Programming Basics with Python - November 2023/JavaScript - Tasks/sumOfTwoNumbers.js
function sumOfTwoNumbers(input) {
    let start = Number(input[0]);
    let end = Number(input[1]);
    let magicNumber = Number(input[2]);
    isFound = false;
    counter = 0;
    let first =  0;
    let second = 0;

    for(let i = start; i <= end; i++){
        if (isFound){
            break;
        }
        for(let j = start; j <= end; j++){
            if (isFound){
                break;
            }
            counter++;
            if(i + j == magicNumber){
                first = i;
                second = j
                isFound = true;
            }  
        }
    }
    isFound? console.log(`Combination N:${counter} (${first} + ${second} = ${magicNumber})`)
    :console.log(`${counter} combinations - neither equals ${magicNumber}`);
}
sumOfTwoNumbers([1,10,5]);